app/src/main/java/com/github/se/eventradar/viewmodel/qrCode/QrCodeAnalyser.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.se.eventradar.viewmodel.qrCode
+
+import android.graphics.ImageFormat
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageProxy
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.BinaryBitmap
+import com.google.zxing.DecodeHintType
+import com.google.zxing.MultiFormatReader
+import com.google.zxing.PlanarYUVLuminanceSource
+import com.google.zxing.common.HybridBinarizer
+import java.nio.ByteBuffer
+import javax.inject.Inject
+
+class QrCodeAnalyser @Inject constructor() : ImageAnalysis.Analyzer {
+
+  // list of supported Image Formats
+  private val supportedImageFormats =
+      listOf(ImageFormat.YUV_420_888, ImageFormat.YUV_422_888, ImageFormat.YUV_444_888)
+
+  var onDecoded: ((String?) -> Unit)? = null //
```

Comment:
Is this still a TODO ?
edit : either comment "Set by the viewModel" or create a Setter function with `private var onDecoded` for more clarity that this need to be set from the outside by the entity that create an instance of QrCodeAnalyser, or even add it as an argument in the constructor of the class (also not sure why you use @Inject on the constructor of that class)

Reply:
you are right about inject, took it off